addon/components/x-option.js

```diff
@@ -55,9 +64,9 @@ export default Ember.Component.extend({
     this._super.apply(this, arguments);
 
     Ember.run.scheduleOnce('afterRender', () => {
-      var select = this.nearestOfType(XSelectComponent);
+      /* Use an assigned select, or search for one nearby */
+      var select = this.get('select') || this.nearestOfType(XSelectComponent);
```

Comment:
Still check for `nearestOfType` to try to avoid breaking anyone that hasn't upgrade to a version of ember that breaks it anyway.